backend/services/syson-services/src/main/java/org/eclipse/syson/util/IDescriptionNameGenerator.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.util;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.sirius.components.diagrams.description.NodeDescription;
+
+/**
+ * Interface of name generator used by all SysON description providers.
+ *
+ * @author Jerome Gout
+ */
+public interface IDescriptionNameGenerator {
```

Comment:
Thank you for the comments on all methods.
I'm not sure that all the `@return` comments are appropriate though.
The interface does not have to tell what should be return. It is the implementation responsibility.
What do you think about?

Reply:
Yes, you are right, I introduced implementation details here which is not the right place. 
Such explanations should rather be in the implementation class `DescriptionNameGenerator`